packages/components/table/src/table-body/index.ts

```diff
@@ -36,7 +36,14 @@ export default defineComponent({
         raf = (fn) => window.setTimeout(fn, 16)
       }
       raf(() => {
-        const rows = instance?.vnode.el?.querySelectorAll(`.${ns.e('row')}`)
+        // just get first level children; fix #9723
+        const el = instance?.vnode.el as HTMLElement
+        const tempId = Date.now()
+        el?.setAttribute('tempId', String(tempId))
+        const rows = el?.querySelectorAll(
+          `[tempId="${tempId}"] > .${ns.e('row')}`
+        )
+        el?.removeAttribute('tempId')
```

Comment:
Is it possible to use the class name directly without using the timestamp here?

Reply:
> Is it possible to use the class name directly without using the timestamp here?

here is the another way to get rows, which also works and passes the test

```js
        const rows = Array.from(el.children).filter((e) =>
          e?.classList.contains(`${ns.e('row')}`)
        )
```

If it's better, I'll commit changes later

Reply:
looks great